newIDE/app/src/ObjectsList/index.js

```diff
@@ -445,8 +445,8 @@ const ObjectsList = React.forwardRef<Props, ObjectsListInterface>(
             objectFolderOrObject
               .getParent()
               .removeFolderChild(objectFolderOrObject);
-            forceUpdateList();
             selectObjectFolderOrObjectWithContext(null);
+            forceUpdateList();
```

Comment:
Should we first set the folder to null, then call `removeFolderChild`, then updateList?
Ideally we'd want to call the update of the list only after everything's been done, but those actions are kinda asynchronous, and we don't really have a way of waiting?
Or you use some kind of effect to call the update only when you're sure everything's done?

Reply:
I moved it up once more, it won't hurt.
I'd rather not use effect for the moment. Maybe if one day we can be sure that's really the root cause

Reply:
Pretty sure we don't want to rely on useEffect. We remove something from the memory, we want to synchrously remove any reference from it in the state so that nothing can use it anymore